wire-library/wire-runtime-swift/src/main/swift/Codable/ProtoMap.swift

```diff
@@ -16,162 +16,73 @@
-}
-
-extension ProtoMap : Hashable where Value : Hashable {
-}
-
-#if swift(>=5.5)
-extension ProtoMap : Sendable where Key : Sendable, Value : Sendable {
-}
-#endif
-
-// MARK: - ProtoMapEnumValues
-
-@propertyWrapper
-public struct ProtoMapEnumValues<
-    Key : Hashable & LosslessStringConvertible,
-    Value : ProtoEnum
-> {
-    public var wrappedValue: [Key: Value]
-
-    public init(wrappedValue: [Key: Value]) {
-        self.wrappedValue = wrappedValue
-    }
-}
-
-extension ProtoMapEnumValues : Encodable {
-    public func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: StringCodingKey.self)
-        for kvp in wrappedValue {
-            let codingKey = StringCodingKey(stringValue: kvp.key.description)
-            try container.encode(kvp.value, forKey: codingKey)
-        }
-    }
-}
-
-extension ProtoMapEnumValues : Decodable {
-    public init(from decoder: Decoder) throws {
-        wrappedValue = [:]
-
-        let container = try decoder.container(keyedBy: StringCodingKey.self)
-        for codingKey in container.allKeys {
-            guard let key = Key(codingKey.stringValue) else {
-                throw ProtoDecoder.Error.unparsableString(type: Key.self, value: codingKey.stringValue)
+            if let valueType = Value.self as? StringDecodable.Type {
+                wrappedValue[key] = try decodeStringEncoded(valueType, from: container, forKey: codingKey)
+            } else if let valueType = Value.self as? ProtoEnum.Type {
+                wrappedValue[key] = try decodeEnum(valueType, from: container, forKey: codingKey)
+            } else {
+                wrappedValue[key] = try container.decode(Value.self, forKey: codingKey)
             }
-
-            wrappedValue[key] = try container.decode(BoxedEnum<Value>.self, forKey: codingKey).value
         }
     }
-}
-
-extension ProtoMapEnumValues : Equatable where Value : Equatable {
-}
-
-extension ProtoMapEnumValues : Hashable where Value : Hashable {
-}
-
-#if swift(>=5.5)
-extension ProtoMapEnumValues : Sendable where Key : Sendable, Value : Sendable {
-}
-#endif
-
-// MARK: - ProtoMapStringEncodedValues
-
-@propertyWrapper
-public struct ProtoMapStringEncodedValues<
-    Key : Hashable & LosslessStringConvertible,
-    Value : StringCodable & Codable
-> {
-    public var wrappedValue: [Key: Value]
-
-    public init(wrappedValue: [Key: Value]) {
-        self.wrappedValue = wrappedValue
-    }
-}
 
-extension ProtoMapStringEncodedValues : Encodable {
-    public func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: StringCodingKey.self)
-        for kvp in wrappedValue {
-            let codingKey = StringCodingKey(stringValue: kvp.key.description)
-            let encodedValue = StringEncoded(wrappedValue: kvp.value)
-            try container.encode(encodedValue, forKey: codingKey)
-        }
+    private func decodeStringEncoded<V: StringDecodable, Key: CodingKey>(
```

Comment:
Should these be `decode(stringEncoded:from:forKey)` to mirror the other `encode`/`decode` methods?

Private, so not a huge deal either way.